dlib/serialize.h

```diff
@@ -1080,10 +1080,23 @@ namespace dlib
         try 
         { 
             unsigned long size;
-            deserialize(size,in); 
+            deserialize(size,in);
+            // Possible bug:
```

Comment:
I'm not worried about this kind of thing.  If someone makes a file that has crazy stuff in it then that's a user bug.

Reply:
Ok, that indirectly means dlib should never be used on untrusted input?

Reply:
Yes, don't call `deserialize()` on inputs you don't trust. There shouldn't ever be arbitrary code execution type of errors, but if someone wants to try and deserialize a 100GB block of data they are free to do that.  But most users would not be happy with the results, although some would.

Reply:
I get your argument regarding allocations, but what about a mismatch on the size of the buffer and what is actually read? If the file is short, shouldn't that be flagged as an error? Maybe the input file has been truncated, and it would be nice to alert the user about that.

Reply:
Most definitely. I think this is already caught by existing code. If it isn’t it should be fixed.